src/core/cgroup.c

```diff
@@ -2384,10 +2384,8 @@ void unit_prune_cgroup(Unit *u) {
         is_root_slice = unit_has_name(u, SPECIAL_ROOT_SLICE);
 
         r = cg_trim_everywhere(u->manager->cgroup_supported, u->cgroup_path, !is_root_slice);
-        if (r < 0) {
-                log_unit_debug_errno(u, r, "Failed to destroy cgroup %s, ignoring: %m", u->cgroup_path);
-                return;
-        }
+        if (r < 0)
+                log_unit_full(u, r==-EBUSY ? LOG_DEBUG : LOG_WARNING, r, "Failed to destroy cgroup %s, ignoring: %m", u->cgroup_path);
```

Comment:
ah, it returns EBUSY in this case, not ENOTEMPTY? if so, sounds good.

tiny nitpick: can you add a space before and after the `==`, please?

Also, can you add a brief comment here explaining the situation, something like this:

```
/* Note that in some cases we cannot remove the cgroup because (possibly delegated) unit processes still 
keep it busy with their own stuff. Accept that, and proceed anyway, pretending it was properly removed, so 
that on the next invocation cycle we set it up properly again, and don't mistakenly assume it's already fully 
realized. */
```

Reply:
Yes, its EBUSY

    buczek@theinternet:~$ journalctl -b|grep destroy|tail -1
    Jun 19 16:02:54 theinternet.molgen.mpg.de systemd[1]: user@0.service: Failed to destroy cgroup /user.slice/user-0.slice/user@0.service, ignoring: Device or resource busy

Thanks, sorry for the cosmetic error, I'll send an update.